### PR TITLE
fix recovery failure

### PR DESCRIPTION
**Pull request by patricklx**
recovery fails because there is still a POD in terminating state.
another issue is that POD is not restarting after slave failed to connect to master. exit 1 is called but somehow docker waits for the background process

### CODE REVIEW

**Comment by smileisak** [comment]
@ianmaddox for review ?

**Comment by adamresson** [comment]
Anxiously awaiting this PR :) 

Do you have confidence @patricklx that this will be stable? It looks like you've fought through a lot of weird timing edge cases.

**Comment by patricklx** [comment]
Wanted to click on comment button....
So, the problem I had first was that we want to use affinity, but then pod can stay in pending status.
Also tested some edge cases. Where master cashes and enough sentinels crash to prevent a failover. In that case I elected a master myself. But I want to add a timeout to give sentinels some time.
The problem is that sentinels need to have a master at startup but there isn't one when it crashed...
Also, I had problems with deployment island now using stateful set.
It somehow didn't wait for pods to be ready until starting next one. Will recheck that again

**Comment by adamresson** [comment]
Also, just as an FYI if you haven't seen it, here's another take on deploying Redis HA in kubernetes using an operator: https://github.com/spotahome/redis-operator

File: redis/alpine/redis-launcher.sh [hunk 1]
```diff
@@ -44,6 +44,7 @@ PORTVAR="${ENV_VAR_PREFIX}MASTER_SVC_SERVICE_PORT"
 HOSTVAR="${ENV_VAR_PREFIX}MASTER_SVC_SERVICE_HOST"
 MASTER_LB_PORT="${!PORTVAR}"
 MASTER_LB_HOST="${!HOSTVAR}"
+QUORUM=${QUORUM:-2}
```
Comment (ianmaddox) [inline redis/alpine/redis-launcher.sh]: Thanks for breaking this out

**Comment by indrekj** [comment]
@ianmaddox could you please also create a tag in quay.io ?

**Comment by patricklx** [comment]
@ianmaddox i still didn't test it with deployment, with Stateful set it works fine.

**Comment by indrekj** [comment]
I'm currently running `latest` in acceptance environment using a deployment with success. Have not tried it in production yet.

@patricklx any chance you can show your stateful set configuration? I'm a bit curious.
